source/framework/vx_producer.c

```diff
@@ -0,0 +1,1336 @@
+/*
+ * Copyright (c) 2024 The Khronos Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <vx_internal.h>
+
+static vx_status set_buffer_status(vx_reference current_ref, producer_buffer_status status, vx_producer producer)
+{
+    uint8_t     buffer_id;
+    vx_bool     found                = vx_false_e;
+    vx_bool     forbidden_transition = vx_false_e;
+    const char* state2string[]       = {"IN_GRAPH", "LOCKED", "FREE"};
+
+    pthread_mutex_lock(&producer->buffer_mutex);
```

Comment:
now that we are in ovx, why don't we use tivx_vx_mutex_t?

Reply:
Will be taken care in the followup PR